pkg/security/tests/process_test.go

```diff
@@ -1535,37 +1534,18 @@ func TestProcessResolution(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	var cmd *exec.Cmd
-	var stdin io.WriteCloser
-	defer func() {
-		if cmd != nil {
-			if stdin != nil {
-				stdin.Close()
-			}
-
-			if err := cmd.Wait(); err != nil {
-				t.Fatal(err)
-			}
-		}
-	}()
-
 	test.WaitSignal(t, func() error {
 		var err error
 
-		args := []string{"multi-open", "/tmp/test-process-resolution", "/tmp/test-process-resolution"}
+		args := []string{"open", "/tmp/test-process-resolution", ";",
```

Comment:
I think It won't work as for this test we need to have the process still running in order to be able to resolve the process cache entries. That is why there was this "pause" mechanism in order to do like:

1. open file
2. getchar (pause)
3. event received
4. check resolution (process still in the caches)
5. write stdin to release everything

The `TestOpenProcessPidDiscarder`  have the same pattern

Reply:
Indeed. I restored the behavior using chained commands.
That being said, the ResolveFromProcfs should fail but it does not as it makes use of the cache. I added a comment about it.